### ISSUE TITLE
Bug: Assertion `orchardWallet.CheckpointNoteCommitmentTree(pindex->nHeight)' failed.

### ISSUE THREAD

**Issue by steven-ecc** [zcash/zcash#5846]
I was running the latest `nu5_private_testnet` and left it mining for a while. When I came back, I had a `Segmentation fault (core dumped)` message.  I did not have coredumps enabled and assumed I had an issue with mining. I restarted the node and received this message on the metrics screen which I now get every time I try start the node:

```
zcashd: wallet/wallet.cpp:2636: void CWallet::IncrementNoteWitnesses(const Consensus::Params &, const CBlockIndex *, const CBlock *, MerkleFrontiers &, bool): Assertion `orchardWallet.CheckpointNoteCommitmentTree(pindex->nHeight)' failed.
Aborted (core dumped)
```

I do have this core dump available.

**Comment by ebfull** [comment]
From the logs:

```
2022-04-07T14:16:42.117017Z TRACE checkpoint{block_height=BlockHeight(1823236)}: rustzcash:wallet Expected checkpoint height 1823353, given 1823236
```

**Comment by daira** [comment]
Closed by #5847.
